src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/BooleanToVisibilityConverter.cs

```diff
@@ -18,6 +18,7 @@ namespace NuGet.PackageManagement.UI
     public class BooleanToVisibilityConverter : IValueConverter
     {
         public bool Inverted { get; set; }
+        public bool HiddenWhenFalse { get; set; }
```

Comment:
i think inverted property inverts the behavior.  so hardcoding "hidden when false" into the property name may not work well when you invert the behavior. 
Perhaps a property of type visibility? defaulting to collapsed, but allowing you to set as Hidden.
named -- HiddenOrCollapsed -- or similar?

Reply:
Help me understand how this logic breaks down by introducing HiddenWhenFalse?
False results in Collapsed or Hidden. Whether or not that's due to being Inverted doesn't change anything.

#### HiddenWhenFalse = false
|Inverted|Input|Result|Behavior|
|---|---|---|---|
|False|False|False|Collapsed|
|False|True|True|Visible|
|True|False|True|Visible|
|True|True|False|Collapsed|


#### HiddenWhenFalse = true
|Inverted|Input|Result|Behavior|
|---|---|---|---|
|False|False|False|**_Hidden_**|
|False|True|True|Visible|
|True|False|True|Visible|
|True|True|False|**_Hidden_**|

Reply:
we talked through these ideas:
```
<BooleanToVisibility x:Key="falseToCollapsed" />

<BooleanToVisibility x:Key="falseToHidden" 
    NonVisualState="Hidden" />

<BooleanToVisibility x:Key="trueToCollapsed" 
    Inverted="true" />

<BooleanToVisibility x:Key="trueToHidden" 
    Inverted="true" NonVisualState="Hidden" />



<BooleanToVisibility x:Key="falseToCollapsed" />

<BooleanToVisibility x:Key="falseToHidden" 
    HiddenOrCollapsed="Hidden" />

<BooleanToVisibility x:Key="trueToCollapsed" 
    Inverted="true" />

<BooleanToVisibility x:Key="trueToHidden" 
    Inverted="true" HiddenOrCollapsed="Hidden" />



<BooleanToVisibility x:Key="falseToCollapsed" />

<BooleanToVisibility x:Key="falseToHidden" 
    HiddenWhenFalse="true" />

<BooleanToVisibility x:Key="trueToCollapsed" 
    Inverted="true" />

<BooleanToVisibility x:Key="trueToHidden" 
    Inverted="true" HiddenWhenFalse="true" />
```